src/Shared/Buffers.MemoryPool/PinnedBlockMemoryPoolMetrics.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+#nullable enable
+
+namespace System.Buffers;
+
+internal sealed class PinnedBlockMemoryPoolMetrics
+{
+    // Note: Dot separated instead of dash.
+    public const string MeterName = "System.Buffers.PinnedBlockMemoryPool";
+
+    private readonly Meter _meter;
+    private readonly UpDownCounter<long> _currentMemory;
+    private readonly Counter<long> _totalAllocatedMemory;
+    private readonly Counter<long> _evictedMemory;
+    private readonly Counter<long> _usageRate;
+
+    public PinnedBlockMemoryPoolMetrics(IMeterFactory meterFactory)
+    {
+        _meter = meterFactory.Create(MeterName);
+
+        _currentMemory = _meter.CreateUpDownCounter<long>(
+            "pinnedblockmemorypool.current_memory",
+            unit: "{bytes}",
+            description: "Number of bytes that are currently pooled by the pinned block memory pool.");
+
+        _totalAllocatedMemory = _meter.CreateCounter<long>(
+           "pinnedblockmemorypool.total_allocated",
+            unit: "{bytes}",
+            description: "Total number of allocations made by the pinned block memory pool.");
+
+        _evictedMemory = _meter.CreateCounter<long>(
+           "pinnedblockmemorypool.evicted_memory",
+            unit: "{bytes}",
+            description: "Total number of bytes that have been evicted.");
+
+        _usageRate = _meter.CreateCounter<long>(
+            "pinnedblockmemorypool.usage_rate",
+            unit: "{bytes}",
+            description: "Rate of bytes rented from the pool."
+            );
```

Comment:
How is this a rate?

Reply:
That's what counters are. It just depends how you view the data.

https://learn.microsoft.com/dotnet/core/diagnostics/metrics-instrumentation#create-a-custom-metric
>For Counter instruments, the convention is that collection tools show the total count and/or the rate at which the count is increasing

Reply:
The runtime doesn't control whether it will be viewed as a rate so the counter name and description shouldn't presume that it will be. I'd suggest naming it based on the thing that is actually being counted. An example similar to this for total bytes allocated from the GC: [RuntimeMetrics.cs](https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/Metrics/RuntimeMetrics.cs,44)